src/serde/tools.rs

```diff
@@ -239,16 +224,16 @@ fn test_serialized_length_from_bytes() {
     );
 
     let e = serialized_length_from_bytes(&[0x8f, 0xff]).unwrap_err();
-    assert_eq!(e.kind(), bad_encoding().kind());
-    assert_eq!(e.to_string(), "bad encoding");
+    assert_eq!(e.kind(), io::ErrorKind::UnexpectedEof);
+    assert_eq!(e.to_string(), "copy terminated early");
```

Comment:
is the standard library only using `UnexpectedEof` for copy operations? Is that why the error message is a bit awkward?

Reply:
It's because of the `copy` call. I agree it's a bit confusing. Maybe I should use `.map` to swap out the error message.